src/Middlewares/RoleOrPermissionMiddleware.php

```diff
@@ -4,13 +4,31 @@
 
 use Closure;
 use Illuminate\Support\Facades\Auth;
+use Spatie\Permission\ClientService;
 use Spatie\Permission\Exceptions\UnauthorizedException;
 
 class RoleOrPermissionMiddleware
 {
     public function handle($request, Closure $next, $roleOrPermission, $guard = null)
     {
         $authGuard = Auth::guard($guard);
+
+        $rolesOrPermissions = is_array($roleOrPermission)
+            ? $roleOrPermission
+            : explode('|', $roleOrPermission);
+
+        // For machine-to-machine Passport clients
+        $bearerToken = $request->bearerToken();
+        if ($bearerToken) {
+            $client = ClientService::getClient($bearerToken);
```

Comment:
https://github.com/spatie/laravel-permission/pull/2467#discussion_r1269482204 can be applied to all middlewares, so the `ClientService` class can be removed
```php
public function handle($request, Closure $next, $roleOrPermission, $guard = null)
{
    $authGuard = Auth::guard($guard);

    $bearerToken = $request->bearerToken();
    if ($bearerToken && method_exists($authGuard, 'client')) {
        $user = $authGuard->client();
    }

    $user = $user ?? $authGuard->user();

    if (! $user) {
        throw UnauthorizedException::notLoggedIn();
    }

    if (! method_exists($user, 'hasAnyRole') || ! method_exists($user, 'hasAnyPermission')) {
        throw UnauthorizedException::missingTraitHasRoles($user);
    }

    $rolesOrPermissions = is_array($roleOrPermission)
        ? $roleOrPermission
        : explode('|', $roleOrPermission);

    if (! $user->canAny($rolesOrPermissions) && ! $user->hasAnyRole($rolesOrPermissions)) {
        throw UnauthorizedException::forRolesOrPermissions($rolesOrPermissions);
    }

    return $next($request);
}
```